pages/vaults/[chainID]/[address].tsx

```diff
@@ -16,9 +18,17 @@ import type {ReactElement} from 'react';
 import type {TYearnVault} from '@common/types/yearn';
 
 function Index({router, vaultData}: {router: NextRouter, vaultData: TYearnVault}): ReactElement {
+	const {address, isActive} = useWeb3();
 	const {safeChainID} = useChainID();
 	const {vaults} = useYearn();
 	const currentVault = useRef<TYearnVault>(vaults[toAddress(router.query.address as string)] as TYearnVault || vaultData);
+	const {refresh} = useWallet();
+
+	useEffect((): void => {
+		if (address && isActive && currentVault?.current?.token?.address) {
+			refresh([{token: toAddress(currentVault.current.address)}]);
```

Comment:
We are giving `currentVault.current.address` to the `toAddress` but checking for the existence of `currentVault?.current?.token?.address`, what am I missing?